improver/cli/generate_metadata_cube.py

```diff
@@ -0,0 +1,144 @@
+@cli.with_output
+def process(
+    *,
+    name="air_pressure_at_sea_level",
+    units=None,
+    spatial_grid="latlon",
+    time="20171110T0400Z",
+    time_period: float = None,
+    frt="20171110T0000Z",
+    ensemble_members: int = 8,
+    leading_dimension: cli.comma_separated_list_of_float = None,
+    leading_dimension_json: cli.inputjson = None,
+    percentile=False,
+    probability=False,
+    spp__relative_to_threshold="above",
+    attributes: cli.inputjson = None,
+    resolution: float = None,
+    domain_corner: cli.comma_separated_list_of_float = None,
+    npoints: int = 71,
+    height_levels: cli.comma_separated_list_of_float = None,
+    height_levels_json: cli.inputjson = None,
+    pressure=False,
+):
+    """ Generate a cube with metadata only.
+
+    Args:
+        name (Optional[str]):
+            Output variable name.
+        units (Optional[str]):
+            Output variable units.
+        spatial_grid (Optional[str]):
+            What type of x/y coordinate values to use.  Permitted values are
+            "latlon" or "equalarea".
+        time (Optional[str]):
+            Single cube validity time. Datetime string of format YYYYMMDDTHHMMZ. If time period given, time is used as the upper time bound.
+        time_period (Optional[float]):
+            The period in hours between the time bounds. This is used to calculate the lower time bound.
+        frt (Optional[str]):
+            Single cube forecast reference time. Datetime string of format YYYYMMDDTHHMMZ.
+        ensemble_members (Optional[int]):
+            Number of ensemble members. Default 8, unless percentile or probability set to True. Will not be used if leading_dimension or leading_dimension_json provided.
+        leading_dimension (Optional[List[float]]):
+            List of realizations, percentiles or thresholds.
+        leading_dimension_json (Optional[Dict]):
+            Dictionary containing a list of realizations, percentiles or thresholds. If both leading_dimension and leading_dimension_json are provided, leading_dimension is used.
+        percentile (Optional[bool]):
+            Flag to indicate whether the leading dimension is percentile values. If True, a percentile cube is created.
+        probability (Optional[bool]):
+            Flag to indicate whether the leading dimension is threshold values. If True, a probability cube is created.
+        spp__relative_to_threshold (Optional[str]):
+            Value of the attribute "spp__relative_to_threshold" which is required for IMPROVER probability cubes.
+        attributes (Optional[Dict]):
+            Dictionary of additional metadata attributes.
+        resolution (Optional[float]):
+            Resolution of grid (metres or degrees).
+        domain_corner (Optional[Tuple[float, float]]):
+            Bottom left corner of grid domain (y,x) (degrees for latlon or metres for equalarea).
+        npoints (Optional[int]):
+            Number of points along a single axis.
```

Comment:
Spatial axis?  Each spatial axis?

Reply:
Updated in commit 002a5ea06131ec1dcf121a78f8c867f0d2a8e081.